src/librustc_trans/trans/intrinsic.rs

```diff
@@ -792,10 +792,33 @@ pub fn trans_intrinsic_call<'a, 'blk, 'tcx>(mut bcx: Block<'blk, 'tcx>,
                                     llresult,
                                     call_debug_location),
 
-        (_, "unchecked_udiv") => UDiv(bcx, llargs[0], llargs[1], call_debug_location),
-        (_, "unchecked_sdiv") => SDiv(bcx, llargs[0], llargs[1], call_debug_location),
-        (_, "unchecked_urem") => URem(bcx, llargs[0], llargs[1], call_debug_location),
-        (_, "unchecked_srem") => SRem(bcx, llargs[0], llargs[1], call_debug_location),
+        (_, c) if c == "unchecked_div" || c == "unchecked_rem" => {
+            use rustc::middle::ty::{TyInt, TyUint};
+            let sty = &arg_tys[0].sty;
+
+            if let &TyInt(_) = sty {
+                if c == "unchecked_div" {
+                    SDiv(bcx, llargs[0], llargs[1], call_debug_location)
+                } else if c == "unchecked_rem" {
+                    SRem(bcx, llargs[0], llargs[1], call_debug_location)
+                } else {
+                    unreachable!()
```

Comment:
I think it's fine to just have the `else` clause fall down to `SRem` and not worry about the `unreachable!` here